.github/workflows/deploy-book.yml

```diff
@@ -0,0 +1,43 @@
+# https://jupyterbook.org/en/stable/publish/gh-pages.html#automatically-host-your-book-with-github-actions
+name: deploy-book
+
+# Only run this when the main branch changes
+on:
+  # Uncomment the 'pull_request' line below to manually re-build Jupyter Book
+  # pull_request:
+  push:
+    branches:
+      - main
+    # If your git repository has the Jupyter Book within some-subfolder next to
+    # unrelated files, you can make this run only if a file within that specific
+    # folder has been modified.
+    paths:
+      - '.github/workflows/deploy-book.yml'
+      - 'book/**'
+
+# This job installs dependencies, builds the book, and pushes it to `gh-pages`
+jobs:
+  deploy-book:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3.0.2
+
+    # Install dependencies
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v3.1.2
+      with:
+        python-version: 3.9
```

Comment:
Will need to change this to use miniconda/mambaforge in a follow up PR, because GMT needs to be installed using conda. But this will suffice for now.